### PR TITLE
feat: deploy CDK sub stacks optionally

### LINKED ISSUE #43: Enable deployment of sub stacks in deploy:stage command
Make it possible to not only deploy `foo-bar` but also `foo-bar/*`

### PR DESCRIPTION
**Pull request by hoegertn**
Fixes #43

### CODE REVIEW

File: src/awscdk/base.ts [hunk 1]
```diff
@@ -366,14 +375,15 @@ ${appCode}
    * and destroying the stack when no longer needed.
    */
   protected createFeatureStage() {
+    const stackId = this.baseOptions.deploySubStacks ? `${this.stackPrefix}-feature ${this.stackPrefix}-feature/*` : `${this.stackPrefix}-feature`;
```
Comment (Lock128) [inline src/awscdk/base.ts:378]: shouldnt this be a global variable initiated in the constructor?

File: src/awscdk/base.ts [hunk 2]
```diff
@@ -347,17 +355,18 @@ ${appCode}
    * comparing changes (diff), and destroying the stack when no longer needed.
    */
   protected createPersonalStage() {
+    const stackId = this.baseOptions.deploySubStacks ? `${this.stackPrefix}-personal ${this.stackPrefix}-personal/*` : `${this.stackPrefix}-personal`;
```
Comment (Lock128) [inline src/awscdk/base.ts:358]: shouldnt this be a global variable initiated in the constructor?

File: src/awscdk/base.ts [hunk 3]
```diff
@@ -382,11 +392,12 @@ ${appCode}
    * @param {DeployStageOptions} stage - The stage to create
    */
   protected createPipelineStage(stage: DeploymentStage) {
+    const stackId = this.baseOptions.deploySubStacks ? `${this.stackPrefix}-${stage.name} ${this.stackPrefix}-${stage.name}/*` : `${this.stackPrefix}-${stage.name}`;
```
Comment (Lock128) [inline src/awscdk/base.ts:395]: shouldnt this be a global variable initiated in the constructor?

File: src/awscdk/base.ts [hunk 4]
```diff
@@ -78,6 +78,14 @@ export interface CDKPipelineOptions {
    */
   readonly stackPrefix?: string;
 
+  /**
+   * If set to true all CDK actions will also include <stackName>/* to deploy/diff/destroy substacks of the main stack.
```
Comment (Lock128) [inline src/awscdk/base.ts:82]: why the concept of / to identify sub-stacks, is that common or mandated ?

**Review by Lock128 — COMMENTED** [review]
i added a few comments

File: src/awscdk/base.ts (hunk 1 above)
Comment (hoegertn) [reply src/awscdk/base.ts:378]: It depends on each stage -feature -personal -dev etc

File: src/awscdk/base.ts (hunk 4 above)
Comment (hoegertn) [reply src/awscdk/base.ts:82]: That happens when you create a new Stack inside another stack instance

```
class MyStack extends Stack {
  constructor(...) {
    super(...)

    new Stack(this, 'something', ...)
  }
}
```

File: src/awscdk/base.ts (hunk 1 above)
Comment (Lock128) [reply src/awscdk/base.ts:378]: Then "thats" the part that goes in here and the only shit you do would be an apply format?

you could even pull this out into a method `getStackId(stage)` instead...

File: src/awscdk/base.ts (hunk 4 above)
Comment (Lock128) [reply src/awscdk/base.ts:82]: Thanks, I didnt know that.

**Review by Lock128 — APPROVED** [review]
Thanks!

**Pull request merged**